### ISSUE TITLE
Reschedule reminders

### ISSUE THREAD

**Issue by ChrisLovering** [python-discord/bot#2045]
Quite often users setup a reminder, and when it gets to that time set another reminder for the same reason. This is usually because the reminder is to get them to do some task, and the user doesn't have time to do that task, so delays it.

We should support this workflow natively within the reminders, rather than having the user create a new reminder.

I propose that when the bot sends the notification for the reminder, the reminder includes some buttons that allows the user to re-schedule the reminder for some set periods. Maybe , 1 hour, 1 day, 1 week and the same period that the original remind was set for.

An example for the last one would be that I set a reminder for 4 hours, so that option would be for another 4 hours.

When the "snoozed" reminder comes up, the bot should link to the original message.

**Comment by ChrisLovering** [comment]
Another thing we could do for this is to have a list of titles, that get progressively more aggressive adn the bot goes through them as you snooze more an more.

For example
Your reminder has arrived!
Hey here's another reminder!
Are you actually going to do it this time?
Go do it now!
etc..

**Comment by Shivansh-007** [comment]
Can't you just do something like this?

![Screenshot from 2022-01-11 21-55-25](https://user-images.githubusercontent.com/69356296/148981711-ae326797-dfc4-48cd-b55a-5c8218f2a504.png)

**Comment by ChrisLovering** [comment]
> Can't you just do something like this?
> 
> Edit: just saw your second comment, this isn't meant for that.
> 
> ![Screenshot from 2022-01-11 21-55-25](https://user-images.githubusercontent.com/69356296/148981711-ae326797-dfc4-48cd-b55a-5c8218f2a504.png)

Sort of, but that requires you to type out the command again, this would just be a quick and easy button. Therefore actually supporting this common workflow within the command.

**Comment by Shivansh-007** [comment]
> Sort of, but that requires you to type out the command again, this proposal would just be a quick and easy button. Therefore actually supporting this common workflow within the command.
>
Agreed, but according to your proposal, this only supports a definite set of possible re-scheduling times, in most of the cases one would need to rewrite the command to have their remind time, say 4hrs (example).

Also, for your second comment, I would love to see something like that even if this doesn't get implemented (we could check if it replies to reminder bot message ig?).

**Comment by ChrisLovering** [comment]
Yea, supporting arbitrary snooze times wouldn't be included in this. Having common ones like 1 hour, 1 day, and the original reminder duration seems to be good enough though.

**Comment by TizzySaurus** [comment]
I'd like to have a go at implementing this.

My current thoughts are having a button on the reminder message for snoozing, which will then display a dropdown of reschedule relative times ("1 hour", "6h hours", "1 day", "1 week" and "1 month" or something). I.e. a system similar to `.epoch`, but with a button triggering the drowdown message.

Thoughts on this? @ChrisLovering @Shivansh-007

**Comment by ChrisLovering** [comment]
Seems good 👍 As long as the subsequent remind message still points to the original message :D

**Comment by Shivansh-007** [comment]
Yeah, that seems good.
